packages/reports/src/reports/editors/simple/SelectorUI.jsx

```diff
@@ -0,0 +1,172 @@
+import React, {useCallback, useEffect, useState} from "react";
+import {ActionIcon, Checkbox, InputWrapper, Select, TextInput} from "@mantine/core";
+import {AiOutlinePlus as PlusIcon} from "react-icons/ai";
+import {HiOutlineTrash as TrashIcon} from "react-icons/hi";
+
+import "./SelectorUI.css";
+
+const SELECTOR_TYPE = {
+  SINGLE: {value: "single", label: "Single"},
+  MULTI: {value: "multi", label: "Multi"}
+};
+
+const DEFAULT_SELECTOR_TYPE = SELECTOR_TYPE.SINGLE.value;
+
+/**
+ * Component for rendering user-friendly simple UI for creating selector-type blocks
+ * and converting a user's selections into code.
+ * @param {import("../SimpleUI").SimpleUIProps} props
+ */
+function SelectorUI(props) {
+
+  const {onChange, locale} = props;
+
+  const [selectorIdentifier, setSelectorIdentifier] = useState("");
+  const [selectorType, setSelectorType] = useState(DEFAULT_SELECTOR_TYPE);
+  const [options, setOptions] = useState([]);
+  const stateFields = [selectorIdentifier, selectorType, options];
+
+  useEffect(() => {
+    setSelectorIdentifier(props.simpleState?.name || "");
```

Comment:
This is a nitpick, but I prefer to destructure props in a single location, so that I don't have to chase around my inline usage of `props.var`. Could `simpleState` be pulled out alongside onChange and locale?

Reply:
You got it

Reply:
Change made.